crates/storage/db/src/implementation/mdbx/mod.rs

```diff
@@ -213,6 +213,56 @@ mod tests {
         assert_eq!(first.1, value, "First next should be put value");
     }
 
+    #[test]
+    fn db_walker() {
+        let db: Arc<Env<WriteMap>> = test_utils::create_test_db(EnvKind::RW);
+
+        // PUT (0, 0), (1, 0), (3, 0)
+        let tx = db.tx_mut().expect(ERROR_INIT_TX);
+        vec![0, 1, 3]
+            .into_iter()
+            .try_for_each(|key| tx.put::<CanonicalHeaders>(key, H256::zero()))
+            .expect(ERROR_PUT);
+        tx.commit().expect(ERROR_COMMIT);
+
+        let missing_key = 2;
+        let tx = db.tx().expect(ERROR_INIT_TX);
+        let mut cursor = tx.cursor_read::<CanonicalHeaders>().unwrap();
+
+        let mut walker =
+            Walker { cursor: &mut cursor, start: None, _tx_phantom: std::marker::PhantomData };
```

Comment:
Don't we have a new method on the walker? If not let's add it

Reply:
i don't think we have it, actually do we need to keep the fields public? Was checking around the usage seems we can make the fields private. 

And it makes sense to me if we just keep internal state of `Walker` private exposing only iteration method.

```
pub struct Walker<'cursor, 'tx, T: Table, CURSOR: DbCursorRO<'tx, T>> {
    /// Cursor to be used to walk through the table.
    pub cursor: &'cursor mut CURSOR,
    /// `(key, value)` where to start the walk.
    pub start: IterPairResult<T>,
    /// Phantom data for 'tx. As it is only used for `DbCursorRO`.
    pub _tx_phantom: PhantomData<&'tx T>,
}
```